proposals/ytt/005-semver/README.md

```diff
@@ -0,0 +1,217 @@
+---
+title: "Semver support in the standard library"
+authors: [ "Max Brauer <mbrauer@vmware.com>" ]
+status: "draft"
+approvers: [ "Dmitriy Kalinin <dkalinin@vmware.com>", "Join Ryan <ryanjo@vmware.com>" ]
+---
+
+# Semver
+
+## Problem Statement
+
+Configuration authors frequently encounter strings which denote the version of a
+software. [Semantic versioning](https://semver.org/) is a widely adopted scheme for versioning
+software. In this proposal the term "semver" is used both for _the_ scheme itself as well _an_ instance of a version.
+
+A semantic version is a string, e.g. `3.2.12`, `0.23.0-beta`, `1.99.2-next+build.5`, which contains structured
+information about the version. Configuration authors may want
+to read a version's constituents, mutate them and compare different versions. For example, when authoring `ytt`
+templates:
+
+* I want to assert that a given string is a well-formed semver.
+* I want to change a semver's representation from string another to data structure, e.g. object or array.
```

Comment:
As an author, for what purpose(s) would I want a data structure?

It _sounds_ like a very helpful thing to do, but what task(s) along the main value stream are made possible/better by having access to a parsed semver?

Reply:
Good point. The point I am trying to make is that a string representation does not allow to access a semver's components. Once you can access its components there's no limit to changing its representation to meet your needs, i.e. JSON, YAML, a map with specific keys, an array, something else entirely.

It's not so much about _"a data structure"_, but about _"easily going beyond string"_.

This is under the premise/assumption that parsing a semver yourself is a high bar.

How about this:
```suggestion
* I want to access a semver's components so that I can easily change its representation.
```